qubesmanager/global_settings.py

```diff
@@ -169,12 +169,20 @@ def __apply_system_defaults__(self):
 
         # default netvm
         if utils.did_widget_selection_change(self.default_netvm_combo):
-            try:
-                self.qubes_app.default_netvm = \
-                    self.default_netvm_combo.currentData()
-            except exc.QubesException as ex:
+            new_default_netvm = self.default_netvm_combo.currentData()
+            if new_default_netvm.property_is_default('netvm'):
                 self.errors.append(
-                    "Failed to set Default NetVM due to {}".format(str(ex)))
+                    'Failed to set Default NetVM: qube {} has its own NetVM '
+                    'set to "default". To avoid cycles, please change NetVM'
+                    'of qube {} to a non-default qube.'.format(
+                        str(new_default_netvm), str(new_default_netvm)))
+            else:
+                try:
+                    self.qubes_app.default_netvm = \
+                        self.default_netvm_combo.currentData()
+                except exc.QubesException as ex:
+                    self.errors.append(
+                        "Failed to set Default NetVM due to {}".format(str(ex)))
```

Comment:
In what situation would this text show up?

Reply:
This can appear in all the "other" situations: permission error, something weird changed while the Settings were opened (e.g. the selected VM stopped providing network for some reason, all sorts of "not sure what happened here"). It can also happen if there's a cycle in NetVMs (qube A has netvm B, qube B has netvm C, qube C has netvm default, and you try to set A as default netvm). All sorts of weirdness :)

Reply:
Ok, how about:

> Cannot set default net qube: {}

Reply:
(I gather that `{}` will be another error message from somewhere else.)

Reply:
yeah, the {} is just... whatever the real error message was, whether PermissionError or something else.